### PR TITLE
test(api): add comprehensive JSON tests for TxOut instances

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Added comprehensive test suite for TxOut JSON serialization and deserialization instances across all Cardano eras, ensuring robust handling of datum types and edge cases. Also exported caseBabbageOnlyOrConwayEraOnwards helper function to the public API for cleaner era-specific conditional logic.
  type:
  - test           # fixes/modifies tests
  - refactoring    # QoL changes
  projects:
  - cardano-api
```

# Context

This PR introduces an extensive test suite to validate the JSON instances for `TxOut` across all Cardano eras and transaction contexts. The tests ensure that JSON serialization and deserialization behave correctly for all datum type variations, including edge cases and error scenarios. Additionally, a helper function for era-specific branching has been exposed in the public API to support cleaner conditional logic when dealing with Babbage vs Conway+ era differences.

Preparation for work on https://github.com/IntersectMBO/cardano-api/issues/926

# How to trust this PR

The test suite provides comprehensive coverage of TxOut JSON behavior:

**Test Coverage Added:**
- **Roundtrip tests**: Validate JSON encode/decode for all eras (Shelley through Conway) in both CtxTx and CtxUTxO contexts
- **Edge case tests**: Handle supplemental datum ambiguity, null field behavior, and ToJSON output validation
- **Error case tests**: Verify proper rejection of conflicting datums, mismatched hashes, partial fields, and invalid data
- **Datum-specific tests**: Cover all datum types (None, Hash, Supplemental, Inline) with dedicated test scenarios

**Test Infrastructure:**
- `Test.Cardano.Api.TxOut.Gen`: Specialized generators for TxOut with specific datum types and invalid JSON scenarios
- `Test.Cardano.Api.TxOut.Helpers`: Utilities for JSON field assertions, parse failure validation, and datum equality checks
- `Test.Cardano.Api.TxOut.Json`: Main test module organizing all test suites
- `Test.Cardano.Api.TxOut.JsonRoundtrip`: Property tests for all era/context combinations
- `Test.Cardano.Api.TxOut.JsonEdgeCases`: Tests for supplemental datum behavior and null field handling
- `Test.Cardano.Api.TxOut.JsonErrorCases`: Validation of error handling for malformed JSON

**API Enhancement:**
- Exported `caseBabbageOnlyOrConwayEraOnwards` from `Cardano.Api.Era` to enable cleaner branching logic for Babbage vs Conway+ era features

To run the new tests:
```bash
cabal test cardano-api-test --test-option='-p "TxOut.Json"'
```

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/gen/Test/Gen/Cardano/Api/TxOut.hs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  ShelleyBasedEraAlonzo ->
+    Gen.choice
+      [ genTxOutWithNoDatum ShelleyBasedEraAlonzo
+      , genTxOutWithDatumHash AlonzoEraOnwardsAlonzo
+      , genTxOutWithSupplementalDatum AlonzoEraOnwardsAlonzo
+      ]
+  ShelleyBasedEraBabbage ->
+    Gen.choice
+      [ genTxOutWithNoDatum ShelleyBasedEraBabbage
+      , genTxOutWithDatumHash AlonzoEraOnwardsBabbage
```
Comment (Jimbo4350) [inline cardano-api/gen/Test/Gen/Cardano/Api/TxOut.hs]: See: https://github.com/IntersectMBO/cardano-api/pull/1040#discussion_r2569891973
